src/jit/gschecks.cpp

```diff
@@ -474,6 +474,9 @@ void Compiler::gsParamsToShadows()
             dst = gtNewOperNode(GT_ADDR, TYP_BYREF, dst);
 
             opAssign = gtNewCpObjNode(dst, src, clsHnd, false);
+#if FEATURE_MULTIREG_STRUCTS
```

Comment:
As I explained offline, struct promotion logic is abstracted out in lvaCanPromoteStructType().  If there any additional types for which we should prevent struct promotion we should add required logic/checks in that method or lvaCanPromoteStructVar() rather than spreading it throughout.  It is not scalable to add this check at every place where a Struct variable is created.

Even in importer, we have such cases for Arm and UNIX struct passing.  Even this needs to be cleaned up once we centralize the logic in lvaCanPromoteStructType().  Please open a work item for the clean up in importer.cpp.